algorithms/scripts/RandomForestClassifier-mans2singh.py

```diff
@@ -0,0 +1,272 @@
+
+# coding: utf-8
+
+# # Create a RandomForestClassifier model to predict TP53 mutation from gene expression data in TCGA
+
+# In[154]:
+
+get_ipython().magic(u'matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[155]:
+
+# We're going to be building a 'TP53' classifier 
+GENE = 'TP53'
+
+
+# In[207]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 5000
+param_fixed = {
+    'min_samples': 100,
+    'min_samples_split': 4,
+}
+param_grid = {
+    'max_depth': [x for x in range(1, 10)],
+    'n_estimators' : [ 10 ** x for x in  range(1,4)]
+}
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+# 
+# *Here is some [information](https://ghr.nlm.nih.gov/gene/TP53) about TP53*
+
+# ## Load Data
+
+# In[208]:
+
+if not os.path.exists('data'):
+    os.makedirs('data')
+
+
+# In[209]:
+
+url_to_path = {
+    # X matrix
+    'https://ndownloader.figshare.com/files/5514386':
+        os.path.join('data', 'expression.tsv.bz2'),
+    # Y Matrix
+    'https://ndownloader.figshare.com/files/5514389':
+        os.path.join('data', 'mutation-matrix.tsv.bz2'),
+}
+
+for url, path in url_to_path.items():
+    if not os.path.exists(path):
+        urllib.request.urlretrieve(url, path)
+
+
+# In[210]:
+
+get_ipython().run_cell_magic(u'time', u'', u"path = os.path.join('data', 'expression.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[211]:
+
+get_ipython().run_cell_magic(u'time', u'', u"path = os.path.join('data', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[212]:
+
+y = Y[GENE]
+
+
+# In[213]:
+
+Y.head(6)
+
+
+# In[214]:
+
+# The Series now holds TP53 Mutation Status for each Sample
+y.head(6)
+
+
+# In[215]:
+
+# Here are the percentage of tumors with NF1
+y.value_counts(True)
+
+
+# ## Set aside 10% of the data for testing
+
+# In[216]:
+
+# Typically, this can only be done where the number of mutations is large enough
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+
+# ## Median absolute deviation feature selection
+
+# In[217]:
+
+def fs_mad(x, y):
+    """    
+    Get the median absolute deviation (MAD) for each column of x
+    """
+    scores = mad(x) 
+    return scores, np.array([np.NaN]*len(scores))
+
+# select the top features with the highest MAD
+feature_select = SelectKBest(fs_mad, k=n_feature_kept)
+
+
+# ## Define pipeline and Cross validation model fitting
+
+# In[218]:
+
+clf = RandomForestClassifier(min_samples_leaf=5, random_state=2)
```

Comment:
I think you want to specify `class_weight='balanced'` [here](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html "sklearn RF").